okio/src/main/java/okio/GzipSource.java

```diff
@@ -68,6 +68,10 @@ public GzipSource(Source source) {
     if (byteCount < 0) throw new IllegalArgumentException("byteCount < 0: " + byteCount);
     if (byteCount == 0) return 0;
 
+    if (source.exhausted()) {
```

Comment:
Full disclosure: I agree with both sides of the argument here. If it were in OkHttp I'd fall on the not support side but only in a pure Okio sense do I think special casing empty might be passable.

I should find out whether you can actually have an empty gzip payload (with header + footer but no data) because if you can then that would probably sway me against this change.